Source/EasyNetQ.ApprovalTests/EasyNetQ.DI.Autofac.approved.txt

```diff
@@ -20,5 +20,12 @@ namespace EasyNetQ.DI.Autofac
         public EasyNetQ.DI.IServiceRegister Register<TService, TImplementation>(EasyNetQ.DI.Lifetime lifetime = 1)
             where TService :  class
             where TImplementation :  class, TService { }
+        public EasyNetQ.DI.IServiceRegister RegisterMore<TService>(TService instance)
```

Comment:
Let's choose good names for `Register`/`RegisterMore`. The former has _replace_ meaning, the latter - _add_.

Reply:
I think Microsoft DI uses Add<>() (for last-registration-wins) and TryAdd<>() (for first-registration-wins), maybe we can adopt those?

Reply:
I thought about `TryAdd`. The fact is that it is not presented in this API. MSDI, I think, has the most widest API set:
1. `Add` - used by `RegisterMore` method now
2. `TryAdd`
3. `Replace` - used by `Register` method now
4. `TryAddEnumerable`

So `Try` pattern is not presented in `IServiceRegister` interface. We can add such methods, but this is not needed (now). Moreover, I am sure that the problem of adapting containers to these methods will arise. Some are able to provide information on registered dependencies, and some are not. I did not change the name `Register` to something else so that there were no extra changes in the PR.

Reply:
Another option - keep only one method `Register` but add `bool append = false` argument for backward source code compatibility. Or `bool override = true`. Which one do you think is better?

Reply:
@Pliner ?